web/packages/teleterm/src/mainProcess/agentRunner/agentRunner.ts

```diff
@@ -140,9 +136,13 @@ export class AgentRunner {
       code: number | null,
       signal: NodeJS.Signals | null
     ) => {
-      // Remove handlers when the process exits.
+      // We don't have to worry about the exit event being emitted after an error event, because
+      // even if that happens, we do want the agent to be updated to the exited state and not remain
+      // in the error state.
+      //
+      // Still, guard against an inverse situation where error would be called after exit. It's
+      // unclear when that would happen, but it doesn't hurt to add this one line.
       process.off('error', errorHandler);
-      process.off('spawn', spawnHandler);
```

Comment:
I don't think the spawn event can fire after the exit event so I removed the handler.